src/ERC4626UpgradeableSafe.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {
+    ERC4626Upgradeable,
+    ERC20Upgradeable,
+    IERC20MetadataUpgradeable
+} from "openzeppelin-upgradeable/contracts/token/ERC20/extensions/ERC4626Upgradeable.sol";
+
+/// @title ERC4626UpgradeableSafe.
+/// @author Morpho Labs.
+/// @custom:contact security@morpho.xyz
+/// @notice ERC4626 Tokenized Vault abstract upgradeable implementation tweaking OZ's implementation to make it safer at initialization.
+abstract contract ERC4626UpgradeableSafe is ERC4626Upgradeable {
+    /* CONSTRUCTOR */
+
+    /// @dev The contract automatically disables initializers when deployed so that nobody can highjack the implementation contract.
+    constructor() {
+        _disableInitializers();
+    }
+
+    /* INITIALIZER */
+
+    function __ERC4626UpgradeableSafe_init(IERC20MetadataUpgradeable asset, uint256 initialDeposit)

```

Comment:
To be honest I think that `IERC20MetadataUpgradeable` is more annoying the anything, wdyt?

Reply:
Yes but let's cast to a more commomn ERC20 interface and not IERC20MetadataUpgradeable which is not commonly used. This one is annoying to use

Reply:
But anyway we can keep this one not a blocker

Reply:
Agreed in this case, but there are other concerns: OZ may force us to use this interface
Btw, it seems the OZ dependency is not up-to-date

Reply:
For [v4.8.3](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/releases/tag/v4.8.3)

Knowing that the 4626 implement has been modified in v4.8.0 and v4.8.1 must everything seems ok. I suggest to move to the v4.8.3 (last release)